src/sentry/notifications/notificationcontroller.py

```diff
@@ -553,3 +553,30 @@ def get_notification_provider_value_for_recipient_and_type(
             return value
 
         return NotificationSettingsOptionEnum.NEVER
+
+    def get_users_for_weekly_reports(self) -> list[int]:
```

Comment:
@snigdhas Seems like we can use a combination of `get_notification_value_for_recipient_and_type` and `get_notification_provider_value_for_recipient_and_type` to make this logic super simple, right?

Reply:
We could, but we'd end up calling `self._get_layered_setting_options` N times for N users, and each call would do iterate over the entire dict. We could store the layered settings in memory instead of computing it but I'm not sure that's necessary